### ISSUE TITLE
Getting “Unsupported destination address” when moving between calendars

### ISSUE THREAD

**Issue by chrender** [Kozea/Radicale#834]
I'm currently trying to move an event from one calendar to another and I'm constantly getting a 502 reply. The log shows an “Unsupported destination address” error. Looking at the code this appears to get thrown when the netloc part of the `HTTP_DESTINATION` isn't equal to `HTTP_HOST`. In this case, the header values are as follows:
```
'HTTP_DESTINATION': '/username/4faa8202-4be3-999f-53dc-d0cd768f4c33/5E9AE458-2252-4858-BAA5-340B6A0F8BB7.ics',
'HTTP_HOST': 'hostname.com:15232',
```
So the `HTTP_DESTINATION` doesn't contain a hostname at all while `HTTP_HOST`is non-empty, resulting in the non-equal-condition to be evaluated as true and radicale reporting the destination address error.

The radicale version used is 2.1.9 and it's running standalone, the client is Fantastical 2.4.10. Is there any way to work around this?

Thanks in advance,
Christoph.

**Comment by chrender** [comment]
I've modified `__init__.py` so that
```
 if to_url.netloc != environ["HTTP_HOST"]
```
becomes
```
 if to_url.netloc and to_url.netloc != environ["HTTP_HOST"]
```
I've never done anything in python or with radicale before, so this may be horribly wrong, but this fixed the problem for me. There's no more 502 response and the new items are created correctly.

**Comment by chrender** [comment]
[radicale-502-patch.txt](https://github.com/Kozea/Radicale/files/2067044/radicale-502-patch.txt)

**Comment by pbiering** [comment]
this should be fixed in 3.1.x by https://github.com/Kozea/Radicale/pull/1271
